flake.nix

```diff
@@ -1,28 +1,77 @@
 {
-  description = "(insert short project description here)";
+  description = "Androsphinx - a SPHINX app for Android.";
 
-  # Nixpkgs / NixOS version to use.
-  inputs.nixpkgs.url = "nixpkgs/nixos-20.09";
+  inputs.nixpkgs = {
+    type = "github";
+    owner = "NixOS";
+    repo = "nixpkgs";
+    ref = "3cadb8b32209d13714b53317ca96ccbd943b6e45";
+  };
 
-  # Upstream source tree(s).
-  inputs.hello-src = {
-    url = "git+https://git.savannah.gnu.org/git/hello.git";
+  inputs.securestring-src = {
+    type = "github";
+    owner = "dnet";
+    repo = "pysecstr";
+    rev = "5d143cffd144378e8d50710de6bc05659f8645fd";
+    flake = false;
+  };
+  inputs.pysodium-src = {
+    type = "github";
+    owner = "stef";
+    repo = "pysodium";
+    rev = "3ffe86a4a2c731a993daac4dc142827201519f03";
     flake = false;
   };
-  inputs.gnulib-src = {
-    url = "git+https://git.savannah.gnu.org/git/gnulib.git";
+  inputs.qrcodegen-src = {
+    type = "github";
+    owner = "nayuki";
+    repo = "QR-Code-generator";
+    rev = "71c75cfeb0f06788ebc43a39b704c39fcf5eba7c";
+    flake = false;
+  };
+  inputs.androsphinx-src = {
+    type = "github";
+    owner = "dnet";
+    repo = "androsphinx";
+    rev = "afcab7478357904d323a70a87d2037f7f56fb2f9";
+    flake = false;
+  };
+  inputs.libsphinx-src = {
+    type = "github";
+    owner = "stef";
+    repo = "libsphinx";
+    rev = "51b0c18c94b645bd7ea3bb21aef623318e0b7939";
+    flake = false;
+  };
+  inputs.pwdsphinx-src = {
+    type = "github";
+    owner = "stef";
+    repo = "pwdsphinx";
+    rev = "f7acc8c9f4e01d44ff3ad65d50e96be337741584";
     flake = false;
   };
 
-  outputs = { self, nixpkgs, hello-src, gnulib-src }:
+  outputs = { self, nixpkgs, securestring-src, pysodium-src, androsphinx-src
+    , libsphinx-src, pwdsphinx-src, qrcodegen-src }:
     let
 
-      # Generate a user-friendly version numer.
-      version = builtins.substring 0 8 hello-src.lastModifiedDate;
+      getVersion = input: builtins.substring 0 7 input.rev;
+      securestring-version = getVersion securestring-src;
+      pysodium-version = getVersion pysodium-src;
+      pwdsphinx-version = getVersion pwdsphinx-src;
+      libsphinx-version = getVersion libsphinx-src;
+      qrcodegen-version = getVersion qrcodegen-src;
+      androsphinx-version = getVersion androsphinx-src;
 
       # System types to support.
       supportedSystems = [ "x86_64-linux" ];
 
+      # Mapping from Nix' "system" to Android's "system".
+      androidSystemByNixSystem = {
+        "x86_64-linux" = "linux-x86_64";
+        "x86_64-darwin" = "darwin-x86_64";
```

Comment:
is this necessary at all? especially as `darwin-` is also contradictive with the name and the  only supported system is `linux`

Reply:
Yes, only Linux is supported. However, I wanted to highlight the fact that some work has been done here. I'll add an explanatory comment.